include/SDL3/SDL_stdinc.h

```diff
@@ -582,14 +582,172 @@ extern "C" {
 #define SDL_stack_free(data)            SDL_free(data)
 #endif
 
+/**
+ * Allocate uninitialized memory.
+ *
+ * The allocated memory returned by this function must be freed with SDL_free().
+ *
+ * If `size` is 0, it will be set to 1.
+ *
+ * If you want to allocate memory aligned to a specific alignment, consider using SDL_aligned_alloc().
+ *
+ * \param size the size to allocate.
+ * \returns a pointer to the allocated memory, or NULL if allocation failed.
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_free
+ * \sa SDL_calloc
+ * \sa SDL_realloc
+ * \sa SDL_aligned_alloc
+ */
 extern SDL_DECLSPEC SDL_MALLOC void * SDLCALL SDL_malloc(size_t size);
+
+/**
+ * Allocate a zero-initialized array.
+ *
+ * The memory returned by this function must be freed with SDL_free().
+ *
+ * If either of `nmemb` or `size` is 0, they will both be set to 1.
+ *
+ * \param nmemb the number of elements in the array.
+ * \param size the size of each element of the array.
+ * \returns a pointer to the allocated array, or NULL if allocation failed.
+ *
+ * \threadsafety It is safe to call this function from any thread.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_free
+ * \sa SDL_malloc
+ * \sa SDL_realloc
+ */
 extern SDL_DECLSPEC SDL_MALLOC SDL_ALLOC_SIZE2(1, 2) void * SDLCALL SDL_calloc(size_t nmemb, size_t size);
+
+/**
+ * Change the size of allocated memory.
+ *
+ * The memory returned by this function must be freed with SDL_free().
+ *
+ * If this function returns a non-NULL pointer,
+ * the `mem` pointer will have been freed and cannot be dereferenced anymore.
+ *
+ * If this function returns NULL (indicating failure),
+ * the `mem` pointer will still be valid and must still be freed with SDL_free().
+ *
+ * If `mem` is NULL, the behavior of this function is equivalent to SDL_malloc().
+ *
+ * If `size` is 0, the behavior of this function is unspecified.
```

Comment:
```suggestion
 * If `size` is 0, it will be set to 1.
```
Note to self that #10756 changed the behavior and made it well-defined.

Reply:
#10756 has been merged, so I think it would be better to apply this change.

Perhaps it would be worth adding something like this (from #10769):

```
 * In some `realloc` implementations, `realloc(mem, 0)` frees `mem`
 * without allocating any memory to replace it. `SDL_realloc` does not
 * do this: `SDL_realloc(mem, 0)` always attempts to allocate and return
 * a memory region, so its behaviour is consistent with `SDL_realloc(mem, 1)`.
```